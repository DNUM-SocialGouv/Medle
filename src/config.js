import getConfig from "next/config"

const { publicRuntimeConfig } = getConfig() || {}

export const START_YEAR_MEDLE = 2020

// Timeout config : keep this timeout values in sync
export const timeout = {
  cookie: 9 * 60 * 60,
  jwt: "9h",
  session: { hours: 9 },
}

export const API_URL = publicRuntimeConfig ? publicRuntimeConfig.API_URL : "http://localhost:3000/api"

export const isOpenFeature = (feature) => {
  const flags = publicRuntimeConfig?.FEATURE_FLAGS || {}
  return !!flags[feature]
}

export const LOGIN_ENDPOINT = "/login"
export const LOGOUT_ENDPOINT = "/logout"
export const RESET_PWD_ENDPOINT = "/reset-password"
export const FORGOT_PWD_ENDPOINT = "/forgot-password"

export const ACTS_ENDPOINT = "/acts"
export const ASKERS_ENDPOINT = "/askers"
export const EMPLOYMENTS_ENDPOINT = "/employments"
export const ATTACKS_ENDPOINT = "/attacks"
export const HOSPITALS_ENDPOINT = "/hospitals"
export const USERS_ENDPOINT = "/users"

export const GLOBAL_STATISTICS_ENDPOINT = "/statistics/global"
export const LIVING_STATISTICS_ENDPOINT = "/statistics/living"
export const DEACEASED_STATISTICS_ENDPOINT = "/statistics/deceased"

export const MESSAGES_ENDPOINT = "/messages"
