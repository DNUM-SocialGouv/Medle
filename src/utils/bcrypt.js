import { compare, hash } from "bcryptjs"

// FaÃ§ade for bcrypt functions
export const hashPassword = async (password) => await hash(password, 10)

export const compareWithHash = async (password, hash) => compare(password, hash)
