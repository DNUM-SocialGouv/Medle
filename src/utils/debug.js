import { useRouter } from "next/router"
import { useEffect, useRef } from "react"

import { logInfo } from "./logger"

// useEffect(() => {
//    logInfo("XX dans useEffect")
//    const routeChangeComplete = async () => {
//       logInfo("XXX dans routeChangeComplete")
//       window.scrollTo(0, 0)

//       logInfo("mainRef", mainRef)

//       if (mainRef && mainRef.current) {
//          mainRef.current.focus()
//       }
//    }
//    router.events.on("routeChangeComplete", routeChangeComplete)

//    return () => {
//       router.events.off("routeChangeComplete", routeChangeComplete)
//    }
// }, [router.events])

export const DebugRouter = () => {
  const router = useRouter()

  router.events.on("routeChangeStart", (url) => {
    logInfo("event routeChangeStart", url)
  })
  router.events.on("routeChangeComplete", (url) => {
    logInfo("event routeChangeComplete", url)
  })
  router.events.on("routeChangeError", (err, url) => {
    logInfo("event routeChangeError", url, err)
  })

  return null
}
// Usage
// function MyComponent(props) {
//    useTraceUpdate(props)
//    return <div>{props.children}</div>
// }
export const useTraceUpdate = (props) => {
  const prev = useRef(props)
  useEffect(() => {
    const changedProps = Object.entries(props).reduce((ps, [k, v]) => {
      if (prev.current[k] !== v) {
        ps[k] = [prev.current[k], v]
      }
      return ps
    }, {})
    if (Object.keys(changedProps).length > 0) {
      logInfo("Changed props:", changedProps)
    }
    prev.current = props
  })
}

// usage: <Logger label="Mon composant Ã  loguer" />
export const Logger = (props) => {
  logInfo(`${props.label} rendered`)
  return null // irrelevant
}
