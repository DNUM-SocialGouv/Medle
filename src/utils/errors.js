import { STATUS_500_INTERNAL_SERVER_ERROR } from "./http"
import { logError } from "./logger"
import { redirectIfUnauthorized } from "./auth"

class MedleError extends Error {
  constructor({ message, detail }) {
    super(message)
    this.name = this.constructor.name
    this.detail = detail
  }
}

export class APIError extends MedleError {
  constructor({ message, detail, status }) {
    super({ message, detail })
    this.name = this.constructor.name
    this.status = status
  }
}
export class InternalError extends MedleError {
  constructor({ detail }) {
    super({ message: "Internal server error", detail })
    this.name = this.constructor.name
    this.status = STATUS_500_INTERNAL_SERVER_ERROR
  }
}

export class ValidationError extends MedleError {
  constructor(message, detail) {
    super({ message, detail })
    this.name = this.constructor.name
  }
}

export const handleAPIResponse = async (response) => {
  if (!response.ok) {
    const json = await response.json()

    const error = new APIError(json)
    logError(error)
    // TODO : problème si SSR et problème 401, car ça devrait redirigé mais il n'y a pas de contexte...
    redirectIfUnauthorized(error)
    return
  }
  return response.json()
}

// TODO : faire une migration progressive sur cette méthode et la renommer à la fin
export const handleAPIResponse2 = async (response) => {
  if (!response.ok) {
    const json = await response.json()

    const { message, detail, status } = json

    const error = new Error(message)
    error.detail = detail
    error.status = status
    logError(error)
    throw error
  }
  return response.json()
}

export const stringifyError = (error) => {
  // eslint-disable-next-line no-unused-vars
  const [stack, ...keys] = Object.getOwnPropertyNames(error)
  return JSON.stringify(error, keys, " ")
}
