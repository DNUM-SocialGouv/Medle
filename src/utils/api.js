import {
   STATUS_200_OK,
   STATUS_401_UNAUTHORIZED,
   STATUS_403_FORBIDDEN,
   STATUS_405_METHOD_NOT_ALLOWED,
   STATUS_500_INTERNAL_SERVER_ERROR,
} from "./http"
import { isAllowed, NO_PRIVILEGE_REQUIRED } from "./roles"
import { checkToken, decodeToken } from "./jwt"
import { APIError } from "./errors"

// see Micro.sendError
export const sendError = (req, res, error) => {
   const statusCode = error.statusCode || error.status
   const message = statusCode ? error.message : "Internal Server Error"

   res.status(statusCode || STATUS_500_INTERNAL_SERVER_ERROR).json({ message })

   if (error instanceof Error) {
      console.error(error.stack)
   } else {
      console.warn("thrown error must be an instance Error")
   }
}

export const sendSuccess = (req, res) => res.status(STATUS_200_OK)

// see Micro.createError
export const createError = (code, message, original) => {
   const error = new Error(message)

   error.statusCode = code
   error.originalError = original

   return error
}

export const sendAPIError = (error, res) => {
   if (error instanceof APIError) {
      return res.status(error.status).json(error)
   }
   // fallback error
   return res
      .status(STATUS_500_INTERNAL_SERVER_ERROR)
      .json({ message: `Erreur de base de donnÃ©e / ${error}`, status: STATUS_500_INTERNAL_SERVER_ERROR })
}

export const checkValidUserWithPrivilege = (privilege, req) => {
   const { token } = req.cookies

   try {
      if (!token) {
         throw new APIError({
            message: "Non authentified user",
            status: STATUS_401_UNAUTHORIZED,
         })
      }

      const currentUser = checkToken(token)

      if (privilege !== NO_PRIVILEGE_REQUIRED && !isAllowed(currentUser.role, privilege)) {
         throw new APIError({
            message: `Not allowed role  (${currentUser.email ? currentUser.email : "unknown user"})`,
            status: STATUS_403_FORBIDDEN,
         })
      } else {
         return currentUser
      }
   } catch (error) {
      let email
      try {
         // Let's try to get user informations even if the token is not valid
         const currentUser = decodeToken(token)
         email = currentUser.email
      } catch (error) {
         console.error("Token couldn't been decoded")
      }
      throw new APIError({
         message: `Invalid token for user (${email ? email : "unknown user"})`,
         status: STATUS_401_UNAUTHORIZED,
      })
   }
}

export const sendMethodNotAllowedError = res => {
   return res.status(STATUS_405_METHOD_NOT_ALLOWED).end()
}
