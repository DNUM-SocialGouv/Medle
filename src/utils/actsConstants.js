import joursFeries from "@socialgouv/jours-feries"
import moment from "moment"

import AsylumSeekerProfile from "../components/profiles/AsylumSeekerProfile"
import BoneAgeProfile from "../components/profiles/BoneAgeProfile"
import CriminalCourtProfile from "../components/profiles/CriminalCourtProfile"
import CustodyProfile from "../components/profiles/CustodyProfile"
import DeceasedProfile from "../components/profiles/DeceasedProfile"
import DrunkProfile from "../components/profiles/DrunkProfile"
import ReconstitutionProfile from "../components/profiles/ReconstitutionProfile"
import RestrainedProfile from "../components/profiles/RestrainedProfile"
import RoadRelatedExaminationProfile from "../components/profiles/RoadRelatedExaminationProfile"
import VictimProfile from "../components/profiles/VictimProfile"
import { ISO_DATE } from "./date"

export const orderedProfileValues = [
  "Victime (vivante)",
  "Gardé.e à vue",
  "Personne pour âge osseux (hors GAV)",
  "Examen pour OFPRA",
  "Personne décédée",
  {
    title: "Autre activité",
    subValues: ["Personne retenue", "Examen lié à la route", "IPM", "Assises", "Reconstitution"],
  },
]

export const profiles = {
  "Victime (vivante)": VictimProfile,
  "Personne décédée": DeceasedProfile,
  "Gardé.e à vue": CustodyProfile,
  "Personne pour âge osseux (hors GAV)": BoneAgeProfile,
  "Examen pour OFPRA": AsylumSeekerProfile,
  "Autre activité/Personne retenue": RestrainedProfile,
  "Autre activité/Examen lié à la route": RoadRelatedExaminationProfile,
  "Autre activité/IPM": DrunkProfile,
  "Autre activité/Assises": CriminalCourtProfile,
  "Autre activité/Reconstitution": ReconstitutionProfile,
}

export const livingProfiles = [
  { value: "Victime (vivante)", label: "Victime" },
  { value: "Gardé.e à vue", label: "Gardé.e à vue" },
  { value: "Personne pour âge osseux (hors GAV)", label: "Personne pour âge osseux" },
  { value: "Examen pour OFPRA", label: "Examen pour OFPRA" },
  { value: "Autre activité/Personne retenue", label: "Personne retenue" },
  { value: "Autre activité/Examen lié à la route", label: "Examen lié à la route" },
  { value: "Autre activité/IPM", label: "IPM" },
]

const periodOfDayValues = {
  week: {
    title: "lun.-ven.",
    period: [
      {
        title: "Nuit profonde",
        subTitle: "(00h-8h30)",
      },
      {
        title: "Journée",
        subTitle: "(8h30-18h30)",
      },
      {
        title: "Soirée",
        subTitle: "(18h30-00h)",
      },
    ],
  },
  saturday: {
    title: "sam.",
    period: [
      {
        title: "Nuit profonde",
        subTitle: "(00h-8h30)",
      },
      {
        title: "Matin",
        subTitle: "(8h30-12h30)",
      },
      {
        title: "Après-midi",
        subTitle: "(12h30-18h)",
      },
      {
        title: "Soirée",
        subTitle: "(18h-00h)",
      },
    ],
  },
  sunday: {
    title: "dim. et férié",
    period: [
      {
        title: "Nuit profonde",
        subTitle: "(00h-8h30)",
      },
      {
        title: "Journée",
        subTitle: "(08h30-18h30)",
      },
      {
        title: "Soirée",
        subTitle: "(18h30-00h)",
      },
    ],
  },
}

const getSituationDate = (dateStr) => {
  const publicHolidays = joursFeries(moment(dateStr).format("YYYY"))

  const publicHolidaysDates = Object.values(publicHolidays).map((elt) => moment(elt).format(ISO_DATE))

  if (publicHolidaysDates.includes(dateStr)) {
    return "sunday"
  }

  const dayInt = moment(dateStr).day()
  switch (dayInt) {
    case 0:
      return "sunday"
    case 6:
      return "saturday"
    default:
      return "week"
  }
}

export { getSituationDate, periodOfDayValues }
