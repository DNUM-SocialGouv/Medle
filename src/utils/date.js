import getConfig from "next/config"
import moment from "moment"

import { logError } from "../utils/logger"

const { publicRuntimeConfig } = getConfig() || {}

export const NAME_MONTHS = {
  "01": "janvier",
  "02": "février",
  "03": "mars",
  "04": "avril",
  "05": "mai",
  "06": "juin",
  "07": "juillet",
  "08": "août",
  "09": "septembre",
  "10": "octobre",
  "11": "novembre",
  "12": "décembre",
}

export const FORMAT_DATE = "DD/MM/YYYY"
export const ISO_DATE = "YYYY-MM-DD"
export const ISO_TIME = "YYYY-MM-DDTHH:mm:ssZ"

export const now = () =>
  (publicRuntimeConfig?.TEST_CURRENT_DATE && moment(publicRuntimeConfig?.TEST_CURRENT_DATE, FORMAT_DATE)) || moment()

export const isValidIsoDate = (date) => date && moment(date, ISO_DATE, true).isValid()

export const frToIso = (date) => {
  const parts = date.split("/")

  if (parts.length !== 3) {
    logError("Problème de parsing de date")
    return date
  }
  const [day, month, year] = parts
  return `${year}-${month}-${day}`
}

export const isoToFr = (date) => {
  const parts = date.split("-")

  if (parts.length !== 3) {
    logError("Problème de parsing de date")
    return date
  }
  const [year, month, day] = parts
  return `${day}/${month}/${year}`
}
