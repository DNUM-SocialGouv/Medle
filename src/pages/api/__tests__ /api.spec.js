import { ASKERS_SEARCH_ENDPOINT, ATTACKS_ENDPOINT, LOGIN_ENDPOINT } from "../../../config"
import { handleAPIResponse } from "../../../utils/errors"
import fetch from "isomorphic-unfetch"

/**
 * Environnement de test :
 * - medle-dev.fabrique.social.gouv.fr
 */

const API_URL = "https://medle-dev.fabrique.social.gouv.fr/api"
const email = "acte@tours.fr"
const password = "test"

let headers

const buildOptionsFetch = async () => {
   const response = await fetch(API_URL + LOGIN_ENDPOINT, {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ email, password }),
   })

   const token = response.headers.get("set-cookie")

   if (!token) {
      throw new Error("Authentication failed")
   }

   headers = {
      headers: {
         cookie: token,
      },
   }
}

describe("endpoints", () => {
   beforeAll(buildOptionsFetch)

   it("should return all attacks for attacks endpoint", async () => {
      const response = await fetch(API_URL + ATTACKS_ENDPOINT, headers)

      const attacks = await handleAPIResponse(response)

      expect(attacks.length).toBe(2)
   })

   it("should return more than 100 commissariats in France for askers endpoint", async () => {
      const response = await fetch(`${API_URL + ASKERS_SEARCH_ENDPOINT}?fuzzy=commissariat&all=true`, headers)

      const askers = await handleAPIResponse(response)
      expect(askers.length).toBeGreaterThan(100)
   })
})
