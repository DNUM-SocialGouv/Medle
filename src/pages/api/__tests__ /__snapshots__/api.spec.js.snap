// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/acts should be possible to an act operator of Tours to add an act 1`] = `
Object {
  "asker": Object {
    "id": 8,
  },
  "examinationDate": "2020-04-14",
  "examinationTypes": Array [
    "Somatique",
    "Psychiatrique",
  ],
  "examinations": Array [
    "Autres",
  ],
  "hospital": Object {
    "id": 1,
  },
  "id": Any<Number>,
  "internalNumber": "wxwxwx123123",
  "location": "UMJ",
  "periodOfDay": "Nuit profonde",
  "personAgeTag": "0-2 ans",
  "personGender": "Féminin",
  "profile": "Victime (vivante)",
  "pvNumber": "",
  "user": Object {
    "id": 2,
  },
  "violenceContexts": Array [
    "Autre type/Autre",
    "Infra-familiale (hors conjugale)",
  ],
  "violenceNatures": Array [
    "Sexuelle",
    "Attentat/Hyper Cacher",
  ],
}
`;

exports[`/askers should return all commissariats in France 1`] = `
Array [
  Object {
    "label": "Section Détachée du Tribunal de Première Instance à Koné",
    "value": 36,
  },
  Object {
    "label": "Section Détachée du Tribunal de Première Instance à Lifou",
    "value": 35,
  },
  Object {
    "label": "Section Détachée du Tribunal de Première Instance à Nuku-Hiva",
    "value": 160,
  },
  Object {
    "label": "Section Détachée du Tribunal de Première Instance à Raïatéa",
    "value": 162,
  },
  Object {
    "label": "Tribunal de grande instance d'Agen",
    "value": 157,
  },
  Object {
    "label": "Tribunal de grande instance d'Aix-en-Provence",
    "value": 117,
  },
  Object {
    "label": "Tribunal de grande instance d'Ajaccio",
    "value": 139,
  },
  Object {
    "label": "Tribunal de grande instance d'Albertville",
    "value": 69,
  },
  Object {
    "label": "Tribunal de grande instance d'Albi",
    "value": 66,
  },
  Object {
    "label": "Tribunal de grande instance d'Alençon",
    "value": 21,
  },
  Object {
    "label": "Tribunal de grande instance d'Alès",
    "value": 81,
  },
  Object {
    "label": "Tribunal de grande instance d'Amiens",
    "value": 121,
  },
  Object {
    "label": "Tribunal de grande instance d'Angers",
    "value": 85,
  },
  Object {
    "label": "Tribunal de grande instance d'Angoulême",
    "value": 63,
  },
  Object {
    "label": "Tribunal de grande instance d'Annecy",
    "value": 111,
  },
  Object {
    "label": "Tribunal de grande instance d'Argentan",
    "value": 22,
  },
  Object {
    "label": "Tribunal de grande instance d'Arras",
    "value": 49,
  },
  Object {
    "label": "Tribunal de grande instance d'Auch",
    "value": 130,
  },
  Object {
    "label": "Tribunal de grande instance d'Aurillac",
    "value": 41,
  },
  Object {
    "label": "Tribunal de grande instance d'Auxerre",
    "value": 134,
  },
  Object {
    "label": "Tribunal de grande instance d'Avesnes-sur-Helpe",
    "value": 6,
  },
  Object {
    "label": "Tribunal de grande instance d'Avignon",
    "value": 24,
  },
  Object {
    "label": "Tribunal de grande instance de Bar-le-Duc",
    "value": 142,
  },
  Object {
    "label": "Tribunal de grande instance de Basse-Terre",
    "value": 124,
  },
  Object {
    "label": "Tribunal de grande instance de Bastia",
    "value": 90,
  },
  Object {
    "label": "Tribunal de grande instance de Bayonne",
    "value": 58,
  },
  Object {
    "label": "Tribunal de grande instance de Beauvais",
    "value": 93,
  },
  Object {
    "label": "Tribunal de grande instance de Belfort",
    "value": 20,
  },
  Object {
    "label": "Tribunal de grande instance de Bergerac",
    "value": 83,
  },
  Object {
    "label": "Tribunal de grande instance de Besançon",
    "value": 131,
  },
  Object {
    "label": "Tribunal de grande instance de Béthune",
    "value": 51,
  },
  Object {
    "label": "Tribunal de grande instance de Béziers",
    "value": 166,
  },
  Object {
    "label": "Tribunal de grande instance de Blois",
    "value": 129,
  },
  Object {
    "label": "Tribunal de grande instance de Bobigny",
    "value": 72,
  },
  Object {
    "label": "Tribunal de grande instance de Bonneville",
    "value": 110,
  },
  Object {
    "label": "Tribunal de grande instance de Bordeaux",
    "value": 37,
  },
  Object {
    "label": "Tribunal de grande instance de Boulogne-sur-Mer",
    "value": 50,
  },
  Object {
    "label": "Tribunal de grande instance de Bourg-en-Bresse",
    "value": 144,
  },
  Object {
    "label": "Tribunal de grande instance de Bourges",
    "value": 170,
  },
  Object {
    "label": "Tribunal de grande instance de Bourgoin-Jallieu",
    "value": 45,
  },
  Object {
    "label": "Tribunal de grande instance de Brest",
    "value": 53,
  },
  Object {
    "label": "Tribunal de grande instance de Brive-la-Gaillarde",
    "value": 153,
  },
  Object {
    "label": "Tribunal de grande instance de Caen",
    "value": 100,
  },
  Object {
    "label": "Tribunal de grande instance de Cahors",
    "value": 147,
  },
  Object {
    "label": "Tribunal de grande instance de Cambrai",
    "value": 9,
  },
  Object {
    "label": "Tribunal de grande instance de Carcassonne",
    "value": 126,
  },
  Object {
    "label": "Tribunal de grande instance de Carpentras",
    "value": 23,
  },
  Object {
    "label": "Tribunal de grande instance de Castres",
    "value": 67,
  },
  Object {
    "label": "Tribunal de grande instance de Cayenne",
    "value": 74,
  },
  Object {
    "label": "Tribunal de grande instance de Châlons-en-Champagne",
    "value": 79,
  },
  Object {
    "label": "Tribunal de grande instance de Chalon-sur-Saône",
    "value": 148,
  },
  Object {
    "label": "Tribunal de grande instance de Chambéry",
    "value": 70,
  },
  Object {
    "label": "Tribunal de grande instance de Charleville-Mézières",
    "value": 16,
  },
  Object {
    "label": "Tribunal de grande instance de Chartres",
    "value": 108,
  },
  Object {
    "label": "Tribunal de grande instance de Châteauroux",
    "value": 146,
  },
  Object {
    "label": "Tribunal de grande instance de Chaumont",
    "value": 102,
  },
  Object {
    "label": "Tribunal de grande instance de Cherbourg-en-Cotentin",
    "value": 137,
  },
  Object {
    "label": "Tribunal de grande instance de Clermont-Ferrand",
    "value": 163,
  },
  Object {
    "label": "Tribunal de grande instance de Colmar",
    "value": 27,
  },
  Object {
    "label": "Tribunal de grande instance de Compiègne",
    "value": 95,
  },
  Object {
    "label": "Tribunal de grande instance de Coutances",
    "value": 138,
  },
  Object {
    "label": "Tribunal de grande instance de Créteil",
    "value": 143,
  },
  Object {
    "label": "Tribunal de grande instance de Cusset",
    "value": 43,
  },
  Object {
    "label": "Tribunal de grande instance de Dax",
    "value": 91,
  },
  Object {
    "label": "Tribunal de grande instance de Dieppe",
    "value": 57,
  },
  Object {
    "label": "Tribunal de grande instance de Digne-les-Bains",
    "value": 65,
  },
  Object {
    "label": "Tribunal de grande instance de Dijon",
    "value": 30,
  },
  Object {
    "label": "Tribunal de grande instance de Douai",
    "value": 10,
  },
  Object {
    "label": "Tribunal de grande instance de Draguignan",
    "value": 113,
  },
  Object {
    "label": "Tribunal de grande instance de Dunkerque",
    "value": 7,
  },
  Object {
    "label": "Tribunal de grande instance de Foix",
    "value": 73,
  },
  Object {
    "label": "Tribunal de grande instance de Fontainebleau",
    "value": 89,
  },
  Object {
    "label": "Tribunal de grande instance de Fort-de-France",
    "value": 104,
  },
  Object {
    "label": "Tribunal de grande instance de Gap",
    "value": 116,
  },
  Object {
    "label": "Tribunal de grande instance de Grasse",
    "value": 13,
  },
  Object {
    "label": "Tribunal de grande instance de Grenoble",
    "value": 47,
  },
  Object {
    "label": "Tribunal de grande instance de Guéret",
    "value": 25,
  },
  Object {
    "label": "Tribunal de grande instance de Laon",
    "value": 106,
  },
  Object {
    "label": "Tribunal de grande instance de La Rochelle",
    "value": 75,
  },
  Object {
    "label": "Tribunal de grande instance de La Roche-sur-Yon",
    "value": 96,
  },
  Object {
    "label": "Tribunal de grande instance de Laval",
    "value": 120,
  },
  Object {
    "label": "Tribunal de grande instance de Libourne",
    "value": 38,
  },
  Object {
    "label": "Tribunal de grande instance de Lille",
    "value": 11,
  },
  Object {
    "label": "Tribunal de grande instance de Limoges",
    "value": 40,
  },
  Object {
    "label": "Tribunal de grande instance de Lisieux",
    "value": 99,
  },
  Object {
    "label": "Tribunal de grande instance de Lons-le-Saunier",
    "value": 68,
  },
  Object {
    "label": "Tribunal de grande instance de Lorient",
    "value": 4,
  },
  Object {
    "label": "Tribunal de grande instance de Lyon",
    "value": 159,
  },
  Object {
    "label": "Tribunal de grande instance de Mâcon",
    "value": 149,
  },
  Object {
    "label": "Tribunal de grande instance de Mamoudzou",
    "value": 155,
  },
  Object {
    "label": "Tribunal de grande instance de Marseille",
    "value": 119,
  },
  Object {
    "label": "Tribunal de grande instance de Meaux",
    "value": 87,
  },
  Object {
    "label": "Tribunal de grande instance de Melun",
    "value": 88,
  },
  Object {
    "label": "Tribunal de grande instance de Mende",
    "value": 128,
  },
  Object {
    "label": "Tribunal de grande instance de Metz",
    "value": 17,
  },
  Object {
    "label": "Tribunal de grande instance de Montargis",
    "value": 1,
  },
  Object {
    "label": "Tribunal de grande instance de Montauban",
    "value": 145,
  },
  Object {
    "label": "Tribunal de grande instance de Montbéliard",
    "value": 132,
  },
  Object {
    "label": "Tribunal de grande instance de Mont-de-Marsan",
    "value": 92,
  },
  Object {
    "label": "Tribunal de grande instance de Montluçon",
    "value": 42,
  },
  Object {
    "label": "Tribunal de grande instance de Montpellier",
    "value": 167,
  },
  Object {
    "label": "Tribunal de grande instance de Moulins",
    "value": 44,
  },
  Object {
    "label": "Tribunal de grande instance de Mulhouse",
    "value": 28,
  },
  Object {
    "label": "Tribunal de grande instance de Nancy",
    "value": 172,
  },
  Object {
    "label": "Tribunal de grande instance de Nanterre",
    "value": 103,
  },
  Object {
    "label": "Tribunal de grande instance de Nantes",
    "value": 32,
  },
  Object {
    "label": "Tribunal de grande instance de Narbonne",
    "value": 127,
  },
  Object {
    "label": "Tribunal de grande instance de Nevers",
    "value": 136,
  },
  Object {
    "label": "Tribunal de grande instance de Nice",
    "value": 14,
  },
  Object {
    "label": "Tribunal de grande instance de Nîmes",
    "value": 80,
  },
  Object {
    "label": "Tribunal de grande instance de Niort",
    "value": 39,
  },
  Object {
    "label": "Tribunal de grande instance de Paris",
    "value": 133,
  },
  Object {
    "label": "Tribunal de grande instance de Pau",
    "value": 59,
  },
  Object {
    "label": "Tribunal de grande instance de Périgueux",
    "value": 82,
  },
  Object {
    "label": "Tribunal de grande instance de Perpignan",
    "value": 77,
  },
  Object {
    "label": "Tribunal de grande instance d'Epinal",
    "value": 60,
  },
  Object {
    "label": "Tribunal de grande instance de Pointe-à-Pitre",
    "value": 123,
  },
  Object {
    "label": "Tribunal de grande instance de Poitiers",
    "value": 84,
  },
  Object {
    "label": "Tribunal de grande instance de Pontoise",
    "value": 140,
  },
  Object {
    "label": "Tribunal de grande instance de Privas",
    "value": 64,
  },
  Object {
    "label": "Tribunal de grande instance de Quimper",
    "value": 52,
  },
  Object {
    "label": "Tribunal de grande instance de Reims",
    "value": 78,
  },
  Object {
    "label": "Tribunal de grande instance de Rennes",
    "value": 150,
  },
  Object {
    "label": "Tribunal de grande instance de Roanne",
    "value": 61,
  },
  Object {
    "label": "Tribunal de grande instance de Rodez",
    "value": 29,
  },
  Object {
    "label": "Tribunal de grande instance de Rouen",
    "value": 55,
  },
  Object {
    "label": "Tribunal de grande instance de Saint-Brieuc",
    "value": 33,
  },
  Object {
    "label": "Tribunal de grande instance de Saint-Denis-de-La-Réunion",
    "value": 114,
  },
  Object {
    "label": "Tribunal de grande instance de Saintes",
    "value": 76,
  },
  Object {
    "label": "Tribunal de grande instance de Saint-Etienne",
    "value": 62,
  },
  Object {
    "label": "Tribunal de grande instance de Saint-Malo",
    "value": 151,
  },
  Object {
    "label": "Tribunal de grande instance de Saint-Nazaire",
    "value": 31,
  },
  Object {
    "label": "Tribunal de grande instance de Saint-Omer",
    "value": 48,
  },
  Object {
    "label": "Tribunal de grande instance de Saint-Pierre",
    "value": 115,
  },
  Object {
    "label": "Tribunal de grande instance de Saint-Quentin",
    "value": 105,
  },
  Object {
    "label": "Tribunal de grande instance de Sarreguemines",
    "value": 18,
  },
  Object {
    "label": "Tribunal de grande instance de Saumur",
    "value": 86,
  },
  Object {
    "label": "Tribunal de grande instance de Saverne",
    "value": 165,
  },
  Object {
    "label": "Tribunal de grande instance de Senlis",
    "value": 94,
  },
  Object {
    "label": "Tribunal de grande instance de Sens",
    "value": 135,
  },
  Object {
    "label": "Tribunal de grande instance de Soissons",
    "value": 107,
  },
  Object {
    "label": "Tribunal de grande instance des Sables-d'Olonne",
    "value": 97,
  },
  Object {
    "label": "Tribunal de grande instance de St Gaudens",
    "value": 169,
  },
  Object {
    "label": "Tribunal de grande instance de Strasbourg",
    "value": 164,
  },
  Object {
    "label": "Tribunal de grande instance de Tarascon",
    "value": 118,
  },
  Object {
    "label": "Tribunal de grande instance de Tarbes",
    "value": 26,
  },
  Object {
    "label": "Tribunal de grande instance de Thionville",
    "value": 19,
  },
  Object {
    "label": "Tribunal de grande instance de Thonon-les-Bains",
    "value": 109,
  },
  Object {
    "label": "Tribunal de grande instance de Toulon",
    "value": 112,
  },
  Object {
    "label": "Tribunal de grande instance de Toulouse",
    "value": 168,
  },
  Object {
    "label": "Tribunal de grande instance de Tours",
    "value": 3,
  },
  Object {
    "label": "Tribunal de grande instance de Troyes",
    "value": 122,
  },
  Object {
    "label": "Tribunal de grande instance de Tulle",
    "value": 154,
  },
  Object {
    "label": "Tribunal de grande instance de Val de Briey",
    "value": 171,
  },
  Object {
    "label": "Tribunal de grande instance de Valence",
    "value": 12,
  },
  Object {
    "label": "Tribunal de grande instance de Valenciennes",
    "value": 8,
  },
  Object {
    "label": "Tribunal de grande instance de Vannes",
    "value": 5,
  },
  Object {
    "label": "Tribunal de grande instance de Verdun",
    "value": 141,
  },
  Object {
    "label": "Tribunal de grande instance de Versailles",
    "value": 54,
  },
  Object {
    "label": "Tribunal de grande instance de Vesoul",
    "value": 71,
  },
  Object {
    "label": "Tribunal de grande instance de Vienne",
    "value": 46,
  },
  Object {
    "label": "Tribunal de grande instance de Villefranche-sur-Saône",
    "value": 158,
  },
  Object {
    "label": "Tribunal de grande instance d'Evreux",
    "value": 125,
  },
  Object {
    "label": "Tribunal de grande instance d'Evry",
    "value": 156,
  },
  Object {
    "label": "Tribunal de grande instance d'Orléans",
    "value": 2,
  },
  Object {
    "label": "Tribunal de grande instance du Havre",
    "value": 56,
  },
  Object {
    "label": "Tribunal de grande instance du Mans",
    "value": 15,
  },
  Object {
    "label": "Tribunal de grande instance du Puy-en-Velay",
    "value": 152,
  },
  Object {
    "label": "Tribunal de Première Instance de Mata-Utu",
    "value": 98,
  },
  Object {
    "label": "Tribunal de Première Instance de Nouméa",
    "value": 34,
  },
  Object {
    "label": "Tribunal de Première Instance de Papeete",
    "value": 161,
  },
  Object {
    "label": "Tribunal de Première Instance de Saint-Pierre-et-Miquelon",
    "value": 101,
  },
]
`;

exports[`/attacks should return all attacks 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Bataclan",
  },
  Object {
    "id": 2,
    "name": "Hyper Cacher",
  },
]
`;
